pkg/accounts/kubernetes/validator.go

```diff
@@ -4,36 +4,166 @@ import (
+	}
+	return nil, noAuthProvidedError
+}
+
+// makeClientFromFile loads the client config from a file path which can be a secret
+func makeClientFromFile(ctx context.Context, file string, settings *authSettings) (clientcmd.ClientConfig, error) {
+	file, err := secrets.DecodeAsFile(ctx, file)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg, err := clientcmd.LoadFromFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientcmd.NewDefaultClientConfig(*cfg, makeOverrideFromAuthSettings(cfg, settings)), nil
+}
+
+// makeClientFromSecretRef reads the client config from a Kubernetes secret in the current context's namespace
+func makeClientFromSecretRef(ctx context.Context, ref *v1alpha2.SecretInNamespaceReference) (clientcmd.ClientConfig, error) {
+	sc, err := secrets.FromContextWithError(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to make kubeconfig file")
+	}
+	str, err := util.GetSecretContent(sc.Client, sc.Namespace, ref.Name, ref.Key)
+	if err != nil {
+		return nil, err
+	}
+	return clientcmd.NewClientConfigFromBytes([]byte(str))
+}
+
+// makeClientFromConfigAPI makes a client config from the v1 Config (the usual format for kubeconfig) inlined
+// into the CRD.
+func makeClientFromConfigAPI(config *clientcmdv1.Config) (clientcmd.ClientConfig, error) {
+	cfg := clientcmdapi.NewConfig()
+	if err := clientcmdlatest.Scheme.Convert(config, cfg, nil); err != nil {
+		return nil, nil
+	}
+	return clientcmd.NewDefaultClientConfig(*cfg, &clientcmd.ConfigOverrides{}), nil
+}
+
+// makeClientFromSettings makes a client config from Spinnaker settings
+func makeClientFromSettings(ctx context.Context, settings map[string]interface{}) (clientcmd.ClientConfig, error) {
+	aSettings := &authSettings{}
+	if err := inspect.Source(aSettings, settings); err != nil {
+		return nil, err
+	}
+	if aSettings.KubeconfigFile != "" {
+		return makeClientFromFile(ctx, aSettings.KubeconfigFile, aSettings)
+	}
+	if aSettings.KubeconfigContents != "" {
+		cfg, err := clientcmd.Load([]byte(aSettings.KubeconfigContents))
+		if err != nil {
+			return nil, err
+		}
+		return clientcmd.NewDefaultClientConfig(*cfg, makeOverrideFromAuthSettings(cfg, aSettings)), nil
+	}
+	return nil, noValidKubeconfigError
+}
+
+func makeOverrideFromAuthSettings(config *clientcmdapi.Config, settings *authSettings) *clientcmd.ConfigOverrides {
+	overrides := &clientcmd.ConfigOverrides{}
+	if settings == nil {
+		return overrides
+	}
+	if settings.Context != "" {
+		overrides.CurrentContext = settings.Context
+	}
+	if settings.User != "" {
+		if authInfo, ok := config.AuthInfos[settings.User]; ok {
+			overrides.AuthInfo = *authInfo
+		}
+	}
+	if settings.Cluster != "" {
+		if cluster, ok := config.Clusters[settings.Cluster]; ok {
+			overrides.ClusterInfo = *cluster
+		}
+	}
+	if len(settings.OAuthScopes) > 0 {
+		overrides.AuthInfo = clientcmdapi.AuthInfo{
+			AuthProvider: &clientcmdapi.AuthProviderConfig{
+				Name: "gcp",
```

Comment:
Is this specific to GCP?

Reply:
Yes. It uses the gcp kubectl plugin. This needs to be validated as part of the Kubernetes validation story.